Exiled.Events/Patches/Generic/Scp173BeingLooked.cs

```diff
@@ -22,12 +29,141 @@ namespace Exiled.Events.Patches.Generic
+    //                        if (!__instance._observingPlayers.Contains(currentPlayerReferenceHub))
+    //                        {
+    //                            __instance._observingPlayers.Add(currentPlayerReferenceHub);
+    //                        }
+    //                    }
+    //                    else if (__instance._observingPlayers.Contains(currentPlayerReferenceHub))
+    //                    {
+    //                        __instance._observingPlayers.Remove(currentPlayerReferenceHub);
+    //                    }
+    //                }
+    //            }
+
+    // __instance._isObserved = __instance._observingPlayers.Count > 0 || __instance.StareAtDuration > 0f;
+    //            if (count != __instance._observingPlayers.Count && __instance._blinkCooldownRemaining > 0f)
+    //            {
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("Adjusting blink cooldown. Initial observers: {0}. ", count) + string.Format("New observers: {0}.", __instance._observingPlayers.Count), MessageImportance.LessImportant, false);
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("Current blink cooldown: {0}", __instance._blinkCooldownRemaining), MessageImportance.LeastImportant, false);
+    //                __instance._blinkCooldownRemaining = Mathf.Max(0f, __instance._blinkCooldownRemaining + ((__instance._observingPlayers.Count - count) * (__instance.BreakneckSpeedsActive ? 0f : 0f)));
+    //                GameCore.Console.AddDebugLog("SCP173", string.Format("New blink cooldown: {0}", __instance._blinkCooldownRemaining), MessageImportance.LeastImportant, false);
+    //                if (__instance._blinkCooldownRemaining <= 0f)
+    //                {
+    //                    __instance.BlinkReady = true;
+    //                }
+    //            }
+    //            HashSetPool<uint>.Shared.Return(hashSet);
+    //            return false;
+    //        }
+    //        catch (Exception generic)
+    //        {
+    //            Log.Error($"Unable to do Scp173BeingLooked {generic}\n");
+    //        }
+
+    // return true;
+    //    }
+    // }
+
     /// <summary>
     /// Patches <see cref="PlayableScps.Scp173.UpdateObservers"/>.
     /// </summary>
     [HarmonyPatch(typeof(PlayableScps.Scp173), nameof(PlayableScps.Scp173.UpdateObservers))]
     internal static class Scp173BeingLooked
     {
+        /// <summary>
+        /// Checks if the current player is to be skipped.
+        /// </summary>
+        /// <param name="instance"> Scp173 instance <see cref="PlayableScps.Scp173"/>. </param>
+        /// <param name="curPlayerGameObj"> Current player game object. </param>
+        /// <param name="curPlayerHub"> Current player referencehub. </param>
+        /// <returns> True if to be skipped, false if not. </returns>
+        public static bool SkipPlayer(ref PlayableScps.Scp173 instance, GameObject curPlayerGameObj, ReferenceHub curPlayerHub)
```

Comment:
Why do you need the `curPlayerGameObj` parameter when you can do `Player.Get(curPlayerHub)`?

Reply:
I don't think you need to use gameObject, when you can do without it
(`Player.Get()` with gameObject is faster, yea, but it's not that critical.)

Reply:
My only other concern is, is there every a possibility that the reference hub, and or game object not match exiled version? (I am curious). With this method, it's impossible to be out of sync because I am providing what NW expects, so if Exiled doesn't have it, they're broken. Otherwise, it should cause no behavior difference. I am willing to remove it but only if we're confident that Exiled will not be something that will have a different result for a referencehub -> game object

Reply:
Also, to get player gameObject, need 2 Instructions:
```cs
new(OpCodes.Ldloca_S, 2),
new(OpCodes.Callvirt, PropertyGetter(typeof(KeyValuePair<GameObject, ReferenceHub>), nameof(KeyValuePair<GameObject, ReferenceHub>.Key))),
```
So removing the gameObject parameter will make the transpiler smaller, but this is subjective, the existence of this parameter will not make it worse

Reply:
If we're confident Exiled will always have a correct matching gameobject to referencehub, I do not mind removing

Reply:
But everything happens :)

Reply:
Alright, I'll update

Reply:
> My only other concern is, is there ever a possibility that the reference hub, and or game object not match exiled version? (I am curious). With this method, it's impossible to be out of sync because I am providing what NW expects, so if Exiled doesn't have it, they're broken. Otherwise, it should cause no behavior difference. I am willing to remove it but only if we're confident that Exiled will not be something that will have a different result for a referencehub -> game object

Honestly, I've never heard of this logic of gameObjects behavior, if it can somehow cause problems in this patch, have Misaka check it out

Reply:
I've made the change, I only hope that it won't happen